extensions/arc/src/ui/dashboards/postgres/postgresExtensionsPage.ts

```diff
@@ -103,56 +112,117 @@ export class PostgresExtensionsPage extends DashboardPage {
-						},
-						async (_progress, _token): Promise<void> => {
-							await this._azdataApi.azdata.arc.postgres.server.edit(
-								this._postgresModel.info.name,
-								{
-									extensions: ''
-								},
-								this._postgresModel.controllerModel.azdataAdditionalEnvVars);
-
-							try {
-								await this._postgresModel.refresh();
-							} catch (error) {
-								vscode.window.showErrorMessage(loc.refreshFailed(error));
+				const addExtDialog = new AddPGExtensionsDialog(this._postgresModel);
+				addExtDialog.showDialog(loc.addExtensions);
+
+				let extArg = await addExtDialog.waitForClose();
+				if (extArg) {
+					try {
+
+						await vscode.window.withProgress(
+							{
+								location: vscode.ProgressLocation.Notification,
+								title: loc.updatingInstance(this._postgresModel.info.name),
+								cancellable: false
+							},
+							async (_progress, _token): Promise<void> => {
+
+
+								await this._azdataApi.azdata.arc.postgres.server.edit(
+									this._postgresModel.info.name,
+									{
+										extensions: this.extensionNames.join() + ',' + extArg
+									},
+									this._postgresModel.controllerModel.azdataAdditionalEnvVars);
+
+								try {
+									await this._postgresModel.refresh();
+								} catch (error) {
+									vscode.window.showErrorMessage(loc.refreshFailed(error));
+								}
 							}
-						}
-					);
+						);
 
-					vscode.window.showInformationMessage(loc.instanceUpdated(this._postgresModel.info.name));
+						vscode.window.showInformationMessage(loc.instanceUpdated(this._postgresModel.info.name));
 
-				} catch (error) {
-					vscode.window.showErrorMessage(loc.instanceUpdateFailed(this._postgresModel.info.name, error));
-				} finally {
-					addExtensionsButton.enabled = true;
+					} catch (error) {
+						vscode.window.showErrorMessage(loc.instanceUpdateFailed(this._postgresModel.info.name, error));
+					}
 				}
 			}));
 
-		return this.modelView.modelBuilder.toolbarContainer().component();
+		return this.modelView.modelBuilder.toolbarContainer().withToolbarItems([
+			{ component: addExtensionsButton }
+		]).component();
 	}
 
 	private refreshExtensionsTable(): void {
+
 		if (this._postgresModel.config) {
-			this.extensions = this._postgresModel.config?.spec.engine.extensions;
-			this.extensionsTable.data = this.extensions.map(e => [e.name]);
+			let extensions = this._postgresModel.config?.spec.engine.extensions;
+			this.extensionsTable.data = extensions.map(e => {
+				this.extensionNames.push(e.name);
+
+				// Can drop individual extensions
+				let dropExtensionsButton = this.modelView.modelBuilder.button().withProps({
+					iconPath: IconPathHelper.delete,
+					ariaLabel: loc.dropExtension,
+					title: loc.dropExtension,
+					width: '20px',
+					height: '20px',
+					enabled: true
+				}).component();
+
+				this.disposables.push(
+					dropExtensionsButton.onDidClick(async () => {
+						try {
+							await vscode.window.withProgress(
```

Comment:
Disable drop while this is running? We may want to even disable running this command more than once since otherwise you have a race condition if you click drop on multiple extensions at a time.